geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegionQueue.java

```diff
@@ -668,4 +679,27 @@ public boolean isReadyForPeek() {
         && !this.eventSeqNumDeque.isEmpty() && getBucketAdvisor().isPrimary();
   }
 
+  List<Object> getQueueList() {
+    getInitializationLock().readLock().lock();
+    try {
+      if (this.getPartitionedRegion().isDestroyed()) {
+        throw new BucketRegionQueueUnavailableException();
+      }
+      List<Object> elementsInQueue = new ArrayList<>();
+      Iterator<Object> it = this.eventSeqNumDeque.iterator();
+      while (it.hasNext()) {
+        Object key = it.next();
+        Object event = optimalGet(key);
+
+        if (!(event instanceof InternalGatewayQueueEvent)) {
+          continue;
+        }
+        elementsInQueue.add(event);
+      }
+      return elementsInQueue;
```

Comment:
Purely personal preference, but this can be simplified significantly using streams:
```
      return eventSeqNumDeque.stream()
          .map(this::optimalGet)
          .filter(o -> o instanceof InternalGatewayQueueEvent)
          .collect(Collectors.toList());
```